engine/language-server/src/main/scala/org/enso/languageserver/search/SuggestionsHandler.scala

```diff
@@ -516,6 +530,18 @@ final class SuggestionsHandler(
         )
       )
 
+    case SuggestionLoadingCompleted =>
+      if (state.suggestionLoadingQueue.nonEmpty) {
+        self ! state.suggestionLoadingQueue.dequeue()
```

Comment:
What does this mean? Send one/all pending requests to self?

Reply:
It deques one message and sends it to self

Reply:
... then the `if` ends and and the actor "become" the same changing `isSuggestionLoadingRunning` state:
- what if the state already was `isSuggestionLoadingRunning`
- what happens sooner? Will the "dequed message" arrive or will we change the state via `become`? Is the order guaranteed?

Reply:
The actor code is single-threaded. What is happening is
- Actor sends the dequeued message to itself
- Actor executes "become" function
- Actor takes the next message from its mailbox (probably the dequeued message that it just sent itself) and starts processing

The actor processes messages in the order they are received. Actor process one message at a time.
This way it's quite easy to write concurrent stuff.